src/HypothesisClient/ApiClient/UsersClient.php

```diff
@@ -52,11 +53,54 @@ public function updateUser(
             Uri::fromParts([
                 'path' => 'users/'.$id,
             ]),
-            $headers,
+            $this->getAuthorizationBasic() + $headers,
             json_encode(array_filter([
                 'email' => $email,
                 'display_name' => $display_name,
             ]))
         );
     }
+
+    public function getUserToken(
+        array $headers,
+        string $id
+    ) : PromiseInterface {
+        return $this->postRequest(
+            Uri::fromParts([
+                'path' => 'token',
+            ]),
+            $headers,
+            json_encode([
+                'form_params' => [
+                    'grant_type' => 'urn:ietf:params:oauth:grant-type:jwt-bearer',
+                    'assertion' => $this->getCredentials()->getJWT($id),
+                ],
+            ])
+        );
+    }
+
+    public function getUserAnnotations(
+        array $headers,
+        string $user,
+        $token = null,
+        int $page = 1,
+        int $perPage = 20,
+        bool $descendingOrder = true,
+        $group = '__world__',
+        $restricted = false
+    ) : PromiseInterface {
+        return $this->getRequest(
+            Uri::fromParts([
+                'path' => 'search',
+                'query' => build_query([
+                    'user' => $user,
+                    'group' => $group,
+                    'offset' => ($page - 1) * $perPage,
+                    'limit' => $perPage,
+                    'order' => $descendingOrder ? 'desc' : 'asc',
+                ]),
+            ]),
+            (($restricted && $token) ? ['Authorization' => 'Bearer '.$token] : []) + $headers
```

Comment:
Is `$restricted` needed?

Reply:
Thinking about this again and I don't like it, we shouldn't be requesting a token if we don't want to see restricted annotations.